test/units/modules/crypto/test_luks_device.py

```diff
@@ -190,39 +221,53 @@ def test_luks_close(device, dev_by_name, name, name_by_dev, state,
                         "get_container_device_by_name",
                         lambda x, y: dev_by_name)
     crypt = luks_device.CryptHandler(module)
-    conditions = luks_device.ConditionsHandler(module, crypt)
-    assert conditions.luks_close() == expected
+    try:
+        conditions = luks_device.ConditionsHandler(module, crypt)
+        assert conditions.luks_close() == expected
+    except ValueError:
+        assert expected == "exception" or expected is False
 
 
-@pytest.mark.parametrize("device, keyfile, new_keyfile, state, expected",
-                         ((d[0], d[1], d[2], d[3], d[4])
+@pytest.mark.parametrize("device, keyfile, new_keyfile, state, label, expected",
+                         ((d[0], d[1], d[2], d[3], d[4], d[5])
                           for d in LUKS_ADD_KEY_DATA))
-def test_luks_add_key(device, keyfile, new_keyfile, state, expected, monkeypatch):
+def test_luks_add_key(device, keyfile, new_keyfile, state, label, expected, monkeypatch):
     module = DummyModule()
     module.params["device"] = device
     module.params["keyfile"] = keyfile
     module.params["new_keyfile"] = new_keyfile
     module.params["state"] = state
+    module.params["label"] = label
+
+    monkeypatch.setattr(luks_device.Handler, "get_device_by_label",
+                        lambda x, y: [0, "/dev/dummy", ""])
 
-    conditions = luks_device.ConditionsHandler(module, module)
     try:
+        conditions = luks_device.ConditionsHandler(module, module)
         assert conditions.luks_add_key() == expected
     except ValueError:
-        assert expected == "exception"
+        assert expected == "exception" or expected is False
 
 
-@pytest.mark.parametrize("device, remove_keyfile, state, expected",
-                         ((d[0], d[1], d[2], d[3])
+@pytest.mark.parametrize("device, remove_keyfile, state, label, expected",
+                         ((d[0], d[1], d[2], d[3], d[4])
                           for d in LUKS_REMOVE_KEY_DATA))
-def test_luks_remove_key(device, remove_keyfile, state, expected, monkeypatch):
+def test_luks_remove_key(device, remove_keyfile, state, label, expected, monkeypatch):
 
     module = DummyModule()
     module.params["device"] = device
     module.params["remove_keyfile"] = remove_keyfile
     module.params["state"] = state
+    module.params["label"] = label
 
-    conditions = luks_device.ConditionsHandler(module, module)
+    monkeypatch.setattr(luks_device.Handler, "get_device_by_label",
+                        lambda x, y: [0, "/dev/dummy", ""])
+    monkeypatch.setattr(luks_device.Handler, "_run_command",
+                        lambda x, y: [0, device, ""])
+
+    crypt = luks_device.CryptHandler(module)
     try:
+        conditions = luks_device.ConditionsHandler(module, crypt)
         assert conditions.luks_remove_key() == expected
     except ValueError:
-        assert expected == "exception"
+        assert expected == "exception" or expected is False
```

Comment:
```suggestion
        assert expected == "exception"
```

Reply:
I was making some tests and forget to clean this line, these tests are based on a lot of "fake mock results"

> It looks like you pushed the wrong version, the current version undoes a lot of things from the previous pushes.

oh man ;/ sorry, let me fix that